bgpd/bgp_evpn_vty.c

```diff
@@ -2484,8 +2491,13 @@ static void evpn_show_all_routes(struct vty *vty, struct bgp *bgp, int type,
 				if (json)
 					json_path = json_object_new_array();
 
-				route_vty_out(vty, &rn->p, pi, 0, SAFI_EVPN,
-					      json_path);
+				if (detail) {
+					route_vty_out_detail(
+						vty, bgp, &rn->p, pi, AFI_L2VPN,
+						SAFI_EVPN, json_path);
+				} else
+					route_vty_out(vty, &rn->p, pi, 0,
```

Comment:
Why '0' is passed for Address Family ?

Reply:
This argument is not AFI. It controls the route display. API was being called with 0 even in the past. I have just moved the call to the else block.

Reply:
what confused me is that, route_vty_out_detail function take AFI and SAFI, where as route_vty_out takes only SAFI and '0'. But you are right. 

Not sure why afi is not required for route_vty_out function though. 

I am good with changes  except the above query. WIll approve

Reply:
Thank you. Let me check on why AFI is not required here. Also, I forgot to add a single line not so important change in the previous commit. Added that now in the new patch.